File: src/Examples/SpecFlowTargetSteps.cs
using System.Collections.Generic;
using TechTalk.SpecFlow;

namespace PB.SpecFlowMaster.Examples
{
    [Binding]
    public class SpecFlowTargetSteps : Steps
    {
        private readonly HashSet<int> _givenExecutionParameters;
        private readonly HashSet<int> _whenExecutionParameters;

        public SpecFlowTargetSteps()
        {
            _givenExecutionParameters = new HashSet<int>();
            _whenExecutionParameters = new HashSet<int>();
        }

        [Given(@"step with parameter (.*)")]
        public void GivenStepWithParameter(int parameter)
        {
            _givenExecutionParameters.Add(parameter);
        }

        [Given(@"step with parameters")]
        public void GivenStepWithParameter(Table table)
        {
            foreach (TableRow row in table.Rows)
            {
                _givenExecutionParameters.Add(int.Parse(row["ParamValue"]));
            }
        }

        [When(@"execute with parameter (.*)")]
        public void WhenExecuteWithParameter(int parameter)
        {
            _whenExecutionParameters.Add(parameter);
        }

        [Then(@"executed Given step with parameter (.*)")]
        public void ThenExecutedGivenStepWithParameter(int parameter)
        {
            AreEqual(true, _givenExecutionParameters.Contains(parameter));
        }

        [Then(@"executed When step with parameter (.*)")]
        public void ThenExecutedWhenStepWithParameter(int parameter)
        {
            AreEqual(true, _whenExecutionParameters.Contains(parameter));
        }
    }
}

File: src/Examples/PB.SpecFlowMaster.Examples.VB/SpecFlowTargetSteps.vb
Imports TechTalk.SpecFlow

' ReSharper disable once CheckNamespace
Namespace PB.SpecFlowMaster.Examples
    <Binding>
    Public Class SpecFlowTargetSteps
        Inherits Steps

        Private ReadOnly _givenExecutionParameters As HashSet(Of Integer)
        Private ReadOnly _whenExecutionParameters As HashSet(Of Integer)

        Public Sub New()
            _givenExecutionParameters = New HashSet(Of Integer)()
            _whenExecutionParameters = New HashSet(Of Integer)()
        End Sub

        <Given("step with parameter (.*)")>
        Public Sub GivenStepWithParameter(ByVal parameter As Integer)
            _givenExecutionParameters.Add(parameter)
        End Sub

        <Given("step with parameters")>
        Public Sub GivenStepWithParameter(ByVal table As Table)
            For Each row As TableRow In table.Rows
                _givenExecutionParameters.Add(Integer.Parse(row("ParamValue")))
            Next
        End Sub

        <[When]("execute with parameter (.*)")>
        Public Sub WhenExecuteWithParameter(ByVal parameter As Integer)
            _whenExecutionParameters.Add(parameter)
        End Sub

        <[Then]("executed Given step with parameter (.*)")>
        Public Sub ThenExecutedGivenStepWithParameter(ByVal parameter As Integer)
            AreEqual(True, _givenExecutionParameters.Contains(parameter))
        End Sub

        <[Then]("executed When step with parameter (.*)")>
        Public Sub ThenExecutedWhenStepWithParameter(ByVal parameter As Integer)
            AreEqual(True, _whenExecutionParameters.Contains(parameter))
        End Sub
    End Class
End Namespace
